src/syntax/delete.ne

```diff
@@ -28,7 +28,7 @@ delete_delete -> (kw_delete %kw_from)
                         });
                     } %}
 
-delete_truncate ->  (kw_truncate %kw_table:?) array_of[table_ref] {% x => track(x, {
+delete_truncate ->  (kw_truncate %kw_table:?) array_of[table_ref] (kw_restart kw_identity):? {% x => track(x, {
                             type: 'truncate table',
                             tables: x[1],
-                        }) %}
```

Comment:
Technically, that parses, but the fact that `restart identity` is present should influence the generated AST.

In another hand, when it is NOT present, the generated AST should be the same as previously.

Moreover, if you implement `restart identity`, you should also implement `continue identity` ... see [the spec](https://www.postgresql.org/docs/current/sql-truncate.html).

Meaning something like: 

```typescript
delete_truncate ->  (kw_truncate %kw_table:?) array_of[table_ref] ((kw_restart | kw_continue) kw_identity):? {% x => track(x, {
                            type: 'truncate table',
                            tables: x[1],
                            ...x[2] && { identity: toStr(x[2][0]) },
                        })
```

and modify the AST definition:

```typescript
export interface TruncateStatement {
    //  [.....]
    identity?: 'restart' | 'continue';
}
```